pynanoleaf/pynanoleaf.py

```diff
@@ -4,8 +4,74 @@
 logger = logging.getLogger(__name__)
 
 
+def _deepgetter(key):
+    """Retrieve value from nested dicts"""
+    def get(data):
+        for k in key:
+            data = data[k]
+        return data
+    return get
+
+
+def _deepsetter(key):
+    """Set value in nested dicts"""
+    def set(data, value):
+        # Fill structure with empty dictionaries up to but not including the
+        # final key part.
+        for k in key[:-1]:
+            data = data.setdefault(k, {})
+        data[key[-1]] = value
+    return set
+
+
+def _dictitem_property(keystr):
+    """Property backed by nested dict items"""
+    key = keystr.split('.')
+    return property(
+        fget=_deepgetter(key),
+        fset=_deepsetter(key)
+    )
+
+
+class Info(dict):
+    """General info about the Nanoleaf"""
+
+    serialNo = _dictitem_property('serialNo')
+    firmwareVersion = _dictitem_property('firmwareVersion')
+    model = _dictitem_property('model')
+    name = _dictitem_property('name')
+
+
+class State(dict):
+    """State of the Nanoleaf"""
+
+    on = _dictitem_property('on.value')
+    brightness = _dictitem_property('brightness.value')
+    min_brightness = _dictitem_property('brightness.min')
+    max_brightness = _dictitem_property('brightness.max')
+    hue = _dictitem_property('hue.value')
+    min_hue = _dictitem_property('hue.min')
+    max_hue = _dictitem_property('hue.max')
+    saturation = _dictitem_property('sat.value')
+    min_saturation = _dictitem_property('sat.min')
+    max_saturation = _dictitem_property('sat.max')
+    color_temperature = _dictitem_property('ct.value')
+    min_color_temperature = _dictitem_property('ct.min')
+    max_color_temperature = _dictitem_property('ct.max')
+    color_mode = _dictitem_property('colorMode')
+
+    def __init__(self, data=None, **kwargs):
+        super().__init__(data or {})
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+
 class Nanoleaf(object):
     """Nanoleaf API wrapper"""
+
+    Info = Info
+    State = State
```

Comment:
Sorry, should've asked that before: Do you want to expose `State` and `Info` to the outside? I'd say they should also be internalized here. (Not the Classes, just the references inside the Nanoleaf object itself)

Reply:
Yeah, or at least State, having this as a "factory method" of sorts on the instance is a nice convenience, both could also be used for type annotations. I was thinking of `pynanoleaf.Nanoleaf.State` (or someinstance.State) as the proper name for these that should be used in code using the module. So I would rather change the module level classes to be `_Info` and `_State`

Reply:
Works for me.  
Please do change the the module wide classes to internal then and I'll cut a new release. And thanks again for the PRs!

Reply:
Done, and thanks for creating the module in the first place :hand: